data_api3/reader.py

```diff
@@ -140,78 +141,78 @@ def read(self, stream):
-
-                        # current_value_extractor = lambda b: numpy.frombuffer(b, dtype=dtype)
-                        # TODO Take care of shape
-                # TODO take care of compression
-
-                # if "shape" in current_channel_info:
-                #     current_shape = current_channel_info["shape"]
-                # else:
-                #     current_shape = None
-
-                # value = None
-                # if current_dtype is not None:
-                #     if current_compression is None:
-                #
-                #         # value = numpy.frombuffer(bytes_read[17:], dtype=current_dtype)
-                #         value = struct.unpack('>d', bytes_read[17:])[0]  # value
-                #     #         if current_shape is not None:
-                #     #             value.reshape(current_shape)
-                #     else:
-                #         # TODO Take care of compression
-                #         value = bytes_read[17:]
-                # else:
-                #     # If type is not supported just save bytes
-                #     if current_compression is None:
-                #         value = bytes_read[17:]
-                #     else:
-                #         # TODO take care of compression
-                #         value = bytes_read[17:]
-
 
+            # Channel header message
+            # A json message that specifies among others data type, shape, compression flags.
+            elif mtype == 0:
+                msg = json.loads(bytes_read[1:])
+                res = process_channel_header(msg)
+                if res.is_error():
+                    logging.error("Can not parse channel header message: {}".format(msg))
+                elif res.is_empty():
+                    logging.info("No data for channel {}".format(res.channel_name))
+                else:
+                    current_channel_info = res.channel_info
+                    current_channel_name = res.channel_name
+                    current_value_extractor = res.value_extractor
+                    current_compression = res.compression
                 current_data = []
                 self.data[current_channel_name] = current_data
 
             bytes_read = stream.read(4)
-            #         length_check = int.from_bytes(bytes_read, byteorder='big')
             length_check = struct.unpack('>i', bytes_read)[0]
             if length_check != length:
                 raise RuntimeError(f"corrupted file reading {length} {length_check}")
 
-        # print(f"{length}, {length_check}")
+
+class ProcessChannelHeaderResult:
+
+    def __init__(self):
+        self.error = False
+        self.empty = False
+        self.channel_info = None
+        self.channel_name = None
+        self.value_extractor = None
+        self.compression = None
+
+    def is_error(self):
+        return self.error
+
+    def is_empty(self):
+        return self.empty
```

Comment:
Why do you have these two function?
The other properties of this class you also directly access (e.g. bla.channel_info)